docs/dev/src/connector/intro.md

```diff
@@ -0,0 +1,192 @@
+
+You write the profile in `risedev.yml` (Or `risedev-profiles.user.yml` ), e.g., the following config includes Kafka and MySQL, which will be used to test sources.
+
+```yml
+  my-cool-profile:
+    steps:
+      # RisingWave cluster
+      - use: minio
+      - use: etcd
+      - use: meta-node
+        meta-backend: etcd
+      - use: compute-node
+      - use: frontend
+      - use: compactor
+      # Connectors
+      - use: kafka
+        address: message_queue
+        port: 29092
+      - use: mysql
+        port: 3306
+        address: mysql
+        user: root
+        password: 123456
+```
+
+Then
+
+```sh
+# will start the cluster along with Kafka and MySQL for you
+risedev d my-cool-profile
+```
+
+For all config options of supported systems, check the comments in `template` section of `risedev.yml` .
+
+### Escape hatch: `user-managed` mode
+
+`user-managed` is a special config. When set to `true` , you will need to start the system by yourself. You may wonder why bother to add it to the RiseDev profile if you start it by yourself. In this case, the config will still be loaded by RiseDev, which will be useful in tests. See chapters below for more details.
+
+The `user-managed` mode can be used as a workaround to start a system that is not yet supported by RiseDev, or is buggy. It's also used to config the CI profile. (In CI, all services are pre-started by `ci/docker-compose.yml` )
+
+Example of the config:
+
+```yml
+      - use: kafka
+        user-managed: true
+        address: message_queue
+        port: 29092
+```
+
+## End-to-end tests
+
+The e2e tests are written in `slt` files. There are 2 main points to note:
+1. Use `system ok` to run `bash` commands to interact with external systems.
+   Use this to prepare the test data, and verify the results. The whole lifecycle of
+   a test case should be written in the same `slt` file.
+2. Use `control substitution on` and then use environment variables to specify the config of the external systems, e.g., the port of Kafka.
+
+Refer to the [sqllogictest-rs documentation](https://github.com/risinglightdb/sqllogictest-rs#extension-run-external-shell-commands) for the details of `system` and `substitution` .
+
+---
+
+Take Kafka as an example about how to the tests are written:
+
+When you use `risedev d` to start the external services, related environment variables for Kafka will be available when you run `risedev slt`:
+
+```sh
+RISEDEV_KAFKA_BOOTSTRAP_SERVERS="127.0.0.1:9092"
+RISEDEV_KAFKA_WITH_OPTIONS_COMMON="connector='kafka',properties.bootstrap.server='127.0.0.1:9092'"
+RPK_BROKERS="127.0.0.1:9092"
+```
+
+The `slt` test case looks like this:
+
+```
+control substitution on
+
+# Note: specifying RPK_BROKERS is unnecessary, since it's already set.
+# Just for demonstrating you can also use envvars in `system` commands.
+system ok
+RPK_BROKERS=$RISEDEV_KAFKA_BOOTSTRAP_SERVERS rpk topic create my_source -p 4
```

Comment:
Can we have a better example here? Still concerned it might be misleading.

Reply:
Can you come up with one?

Reply:
Failed to find any other existing examples in our codebase.

Reply:
Let me just remove the example, and only add a note